pandas/core/dtypes/cast.py

```diff
@@ -996,7 +1018,9 @@ def astype_nansafe(arr, dtype, copy: bool = True, skipna: bool = False):
     return arr.view(dtype)
 
 
-def maybe_convert_objects(values: np.ndarray, convert_numeric: bool = True):
+def maybe_convert_objects(
+    values: np.ndarray, convert_numeric: bool = True
+) -> Union[np.ndarray, DatetimeIndex]:
```

Comment:
Getting a circular import here with `DatetimeIndex`. 

Will revert to `ABCDatetimeIndex` for now and circle back at the end

Reply:
actually nvm https://github.com/pandas-dev/pandas/pull/36842#pullrequestreview-506058423 solves this